### PR TITLE
chore: component version testing workflows and scripts

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

File: .github/describe_supported_versions_diff.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+set -eu
+
+readonly SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+readonly ROOT_DIR="$(dirname ${SCRIPT_DIR})"
+
+echo "feat: Test additional package versions [auto-test-update]"
+
+git diff --name-only -- "${ROOT_DIR}/instrumentation/" | \
+    sort | \
+    while read -r modified_version_file; do \
```
Comment (GuyMoses) [inline .github/describe_supported_versions_diff.sh:12]: maybe this line should be described a bit/ documented? to be more readable

File: .github/gatherVersionArtifacts.sh [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    origin_path="$2"
+    shift 2
+    runtimes=("$@")
+
+    echo "working on: $instrumentation_name"
+
+    declare -A runtime_to_tested_versions=()
+    calculate_runtime_to_tested_versions "$instrumentation_name" "${runtimes[@]}"
+
+# TODO This next section is not working!!
```
Comment (GuyMoses) [inline .github/gatherVersionArtifacts.sh:71]: it it fixable?

**Review by GuyMoses — APPROVED** [review]

File: .github/gatherVersionArtifacts.sh (hunk 2 above)
Comment (kenfinnigan) [reply .github/gatherVersionArtifacts.sh:71]: Oops, forgot to remove the TODO!

File: .github/describe_supported_versions_diff.sh (hunk 1 above)
Comment (kenfinnigan) [reply .github/describe_supported_versions_diff.sh:13]: Added a comment

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**